### PR TITLE
Disconnected <a> should not navigate

### PR DESCRIPTION
**Pull request by annevk**
Follow-up test for https://github.com/whatwg/html/pull/2613.

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5759/)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/227991328)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 716c7e11578227829aef3582f549a0270a28a9b8
Using browser at version BuildID 20170501100238; SourceStamp 076a7a66096f9e8d102548397254be32eb26bc3d
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5759/html/semantics/links/following-hyperlinks/activation-behavior.window.html">/html/semantics/links/following-hyperlinks/activation-behavior.window.html</a></summary>


|                            Subtest                             | Results |                   Messages                   |
|----------------------------------------------------------------|---------|----------------------------------------------|
|                                                                | OK      |                                              |
| `<a> that is not connected should still be followed`           | FAIL    | `assert_unreached: Reached unreachable code` |
| `<a> that is from an inactive document should not be followed` | PASS    |                                              |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/227991334)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 969f828f3bad022d2f9ba2ae2a6257b71522410b

**Comment by annevk** [comment]
As described in https://github.com/whatwg/html/pull/2613 I'm flipping the pass condition.

**Comment by annevk** [comment]
We should remove the second async_test now the pass condition is flipped I think. Will leave it there for now until I get some review.

**Comment by annevk** [comment]
Time to flip the pass condition back again, but only for `<a>`.

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17703)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/267597417)

**Review by jdm — APPROVED** [review]

File: html/semantics/links/following-hyperlinks/activation-behavior.window.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const followed = type === "a" ? true : false;
+  async_test(t => {
+    const target = document.createElement("iframe"),
+          link = document.createElement(type);
+    t.add_cleanup(() => target.remove());
+    target.name = "certifiedrandom" + type;
+    link.target = "certifiedrandom" + type;
+    link.href = "/";
+    document.body.appendChild(target);
+    target.onload = t.step_func(() => {
```
Comment (domenic) [inline html/semantics/links/following-hyperlinks/activation-behavior.window.js]: This test will pass after 500 ms if followed is true and the target never loads. That seems wrong?

File: html/semantics/links/following-hyperlinks/activation-behavior.window.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+["a",
+ "area"].forEach(type => {
```
Comment (domenic) [inline html/semantics/links/following-hyperlinks/activation-behavior.window.js]: Missing "link"

Also isn't this file partially redudant with active-document.window.js, introduced in https://github.com/w3c/web-platform-tests/pull/5758 ?

Comment (annevk) [reply html/semantics/links/following-hyperlinks/activation-behavior.window.js]: The `link` element doesn't have `target`.

Comment (annevk) [reply html/semantics/links/following-hyperlinks/activation-behavior.window.js]: I don't think it's redundant. That's a link in a document that's navigated away. Here it's a link that's simply not connected (or from a synthetic document).

File: html/semantics/links/following-hyperlinks/activation-behavior.window.js (hunk 1 above)
Comment (annevk) [reply html/semantics/links/following-hyperlinks/activation-behavior.window.js]: Good, seems that found a bug in Fx.

**Review by domenic — APPROVED** [review]

**Pull request merged**